src/components/anti-executable.ts

```diff
@@ -37,6 +37,40 @@ export default class AntiExecutable extends BotComponent {
         Buffer.from([0xca, 0xfe, 0xba, 0xbe]), // mach-o universal
         Buffer.from([0xbe, 0xba, 0xfe, 0xca]), // mach-o universal swapped
         Buffer.from([0x4d, 0x5a]), // DOS MZ / PE
+        // Source for the archives and compressed files magic values:
+        //  https://en.wikipedia.org/wiki/List_of_file_signatures
+        Buffer.from([0x1f, 0x9d]), // Compressed file using Lempel-Ziv-Welch algorithm.
+        Buffer.from([0x1f, 0xa0]), // Compressed file using LZH algorithm
+        Buffer.from([0x1f, 0x8b]), // GZIP compressed files
+        Buffer.from([0x78, 0xf9]), // ZLIB (Best compression, with preset dictionary)
+        Buffer.from([0x78, 0x01]), // ZLIB (No Compression)
+        Buffer.from([0x78, 0x5e]), // ZLIB (Best Speed, no preset dictionary)
+        Buffer.from([0x78, 0x9c]), // ZLIB (Default compresson, no preset dictionary)
+        Buffer.from([0x78, 0xda]), // ZLIB (Best compression, no preset dictionary)
+        Buffer.from([0x78, 0x20]), // ZLIB (No compression, with preset dictionary)
+        Buffer.from([0x78, 0x7d]), // ZLIB (Best speed, with preset dictionary)
+        Buffer.from([0x78, 0xbb]), // ZLIB (Default compression, with preset dictionary)
+        Buffer.from([0x1a, 0x08]), // ARC archive file
+        Buffer.from([0x4F, 0x41, 0x52]), // OAR file archive
+        Buffer.from([0x2d, 0x68, 0x6c, 0x30, 0x2d]), // Lempel Ziv Huffman archive file method 0 (No compression)
+        Buffer.from([0x2d, 0x68, 0x6c, 0x35, 0x2d]), // Lempel Ziv Huffman archive file method 5 (8KiB sliding window)
```

Comment:
One small issue here is that currently these signatures are only checked at offset 0, and these two start at index 2. We'd need to encode the non-zero offset somehow for this one.

Reply:
> I've got an idea for encoding files where magic values are at non-zero positions. We can simply add the offset value in the array of magic numbers, for example entry for file can look like this:
```js
[length, Buffer.from([0x01, 0x02, ...])]
```

How's the idea?

Reply:
Perfect!